src/python/TaskWorker/Actions/DBSDataDiscovery.py

```diff
@@ -145,6 +145,16 @@ def requestTapeRecall(self, blockList, sizeToRecall, system='Rucio', msgHead='')
             # make RSEs lists
             # asking for ddm_quota>0 gets rid also of Temp and Test RSE's
             ALL_RSES = "ddm_quota>0&(tier=1|tier=2)&rse_type=DISK"
+            # tune list according to where tapes are, we expect a single location
+            # if there are more... let Rucio deal with them to find best route
+            if tapeLocations:
+                if 'T1_RU_JINR_Tape' in tapeLocations and len(tapeLocations) > 1:
+                    tapeLocations.remove('T1_RU_JINR_Tape')  # JINR tape data are often duplicated
+            if len(tapeLocations) == 1:
+                if 'US' in tapeLocations:
+                    ALL_RSES += "&(country=US|country=BR)"
+                else:
+                    ALL_RSES += "\country=US\country=BR"  # note the set complement operator \
```

Comment:
I would indent this by one level, effectively protecting it with `if tapeLocation:`, because if `tapeLocation` is the default `None`, then `len(None)` will raise an unnecessary "TypeError: object of type 'NoneType' has no len()".

Reply:
thanks Dario ! I think  had the indent, then removed it in a moment of brain loss :-)